docs/user-manual/proxy-protocol.adoc

```diff
@@ -0,0 +1,33 @@
+= PROXY Protocol
+:idprefix:
+:idseparator: -
+:docinfo: shared
+
+As noted in the official https://github.com/haproxy/haproxy/blob/master/doc/proxy-protocol.txt[PROXY Protocol documentation]:
+
+[quote,]
+____
+The PROXY protocol provides a convenient way to safely transport connection information such as a client's address across multiple layers of NAT or TCP proxies.
+____
+
+This essentially allows the broker to know a client's IP address even when the connection is established through reverse proxy that supports the PROXY protocol (e.g. HAProxy, nginx, etc.).
+Without PROXY protocol support the broker would see such client connections as coming from the proxy itself which can be misleading for administrators and complicate trouble-shooting.
+
+Both versions 1 & 2 of the PROXY Protocol are supported.
+Furthermore, this support is 100% transparent and requires no additional configuration.
+The broker automatically detects the use of the PROXY Protocol and manages the connection appropriately.
```

Comment:
This is actually expressly forbidden by the PROXY protocol because it causes a security hole. It essentially has to be configured on so that only brokers that want to accept the PROXY protocol do, so people cant spoof their address. The protocol also requires that it be implemented such that connecting 'clients' are _required_ to use the protocol: https://github.com/haproxy/haproxy/blob/b167d545cf4b673de3c481088d7ce8ed65030106/doc/proxy-protocol.txt#L176-L182.

Reply:
That makes sense.